src/platforms/ctxlink/WiFi_Server.c

```diff
@@ -0,0 +1,1511 @@
+
+static uint32_t timeout_seconds = 0;      // Used to perform a countdown in seconds for app tasks
+static uint32_t timeout_tick_counter = 0; // counts Timer1 ticks (1mS) for an active timeout
+static bool timeout = false;              // Asserted at end of timeout
+
+typedef enum server_state {
+	sm_home = 0,  ///< State of the TCP server
+	sm_listening, ///< An enum constant representing the sm listening option
+	sm_closing,   ///< An enum constant representing the sm closing option
+	sm_idle,      ///< An enum constant representing the sm idle option
+} server_state_e;
+
+static server_state_e gdb_tcp_server_state = sm_idle;
+static server_state_e uart_debug_tcp_server_state = sm_idle;
+static server_state_e swo_trace_tcp_serverstate = sm_idle;
+
+struct sockaddr_in gdb_addr = {0};
+
+struct sockaddr_in uart_debug_addr = {0};
+struct sockaddr_in swo_trace_addr = {0};
+
+static volatile SOCKET socket_parameter = 0;
+
+//
+// Flag used to run the MODE LED state machine
+//
+static bool run_mode_led_task = false; ///< True to run mode LED task
+static uint32_t press_timer = 0;       ///< The press timer
+
+static bool driver_init_complete = false;     ///< True to driver initialize complete
+static bool g_wifi_connected = false;         ///< True if WiFi connected
+static bool ip_address_assigned = false;      ///< True if IP address assigned
+static bool dns_resolved = false;             ///< True if DNS resolved
+static bool gdb_client_connected = false;     ///< True if client connected
+static bool gdb_server_is_running = false;    ///< True if server is running
+static bool new_gdb_client_connected = false; ///< True if new client connected
+
+static SOCKET gdb_server_socket = SOCK_ERR_INVALID; ///< The main gdb server socket
+static SOCKET gdb_client_socket = SOCK_ERR_INVALID; ///< The gdb client socket
+
+#define UART_DEBUG_INPUT_BUFFER_SIZE 32
+static uint8_t local_uart_debug_buffer[UART_DEBUG_INPUT_BUFFER_SIZE] = {0}; ///< The local buffer[ input buffer size]
+
+static SOCKET uart_debug_server_socket = SOCK_ERR_INVALID;
+static SOCKET uart_debug_client_socket = SOCK_ERR_INVALID;
+static bool uart_debug_client_connected = false;
+static bool user_configured_uart = false;
+static bool uart_debug_server_is_running = false;
+static bool new_uart_debug_client_conncted = false;
+
+static SOCKET swo_trace_server_socket = SOCK_ERR_INVALID;
+static SOCKET swo_trace_client_socket = SOCK_ERR_INVALID;
+static bool swo_trace_client_connected = false;
+static bool swo_trace_server_is_running = false;
+static bool new_swo_trace_client_conncted = false;
+
+#define SWO_TRACE_INPUT_BUFFER_SIZE 32
+static uint8_t local_swo_trace_buffer[SWO_TRACE_INPUT_BUFFER_SIZE] = {0}; ///< The local buffer[ input buffer size]
+
+#define WPS_LOCAL_TIMEOUT 30 // Timeout value in seconds
+
+//
+// Sign-on message for new UART data clients
+//
+static char uart_client_signon[] = "\r\nctxLink UART connection.\r\nPlease enter the UART setup as baud, bits, parity, "
```

Comment:
You may wish to mark this `const` so it doesn't get copied from Flash to RAM every start-up and then consume space in both. eg `static const char *uart_client_signon` or `static const char uart_client_signon[]` - either works, though the former is our preference unless the length of the array is required.

Reply:
Addressed. The string length is used later in the module, hence the chosen format.